src/libraries/System.Private.CoreLib/src/System/SearchValues/ProbabilisticWithAsciiCharSearchValues.cs

```diff
@@ -42,7 +42,7 @@ internal override int IndexOfAny(ReadOnlySpan<char> span)
 
             // We check whether the first character is ASCII before calling into IndexOfAnyAsciiSearcher
             // in order to minimize the overhead this fast-path has on non-ASCII texts.
-            if (IndexOfAnyAsciiSearcher.IsVectorizationSupported && span.Length >= Vector128<short>.Count && char.IsAscii(span[0]))
```

Comment:
I think we need to keep `IsSupported` checks to ensure trimming works as expected on platforms without SIMD support.

Reply:
Is that needed even though all its ctor uses are guarded behind it?

Reply:
R2R has some special rules for various ISA checks and I'm pretty sure we expect them to be used even when all callers are expected to hit a given path.

CC. @davidwrighton